module/documents/actor/actor.mjs

```diff
@@ -356,6 +356,64 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
     });
   }
 
+  /* -------------------------------------------- */
+
+  /**
+   * Deduct a certain amount of currency from this Actor.
+   * @param {number} amount                          The amount of currency.
+   * @param {string} denomination                    The currency's denomination.
+   * @param {object} [options]
+   * @param {boolean} [options.recursive=false]      Deduct currency from containers as well as the base Actor. TODO
+   * @param {"high"|"low"} [options.priority="low"]  Prioritize higher denominations before lower, or vice-versa.
+   * @param {boolean} [options.exact=true]           Prioritize deducting the requested denomination first.
+   * @throws {Error} If the Actor does not have sufficient currency.
+   * @returns {Promise<Actor5e>|void}
+   */
+  deductCurrency(amount, denomination, options={}) {
```

Comment:
We should probably implement these on the `CurrencyManager` application which also houses `convertCurrency` and `transferCurrency` implementations. That would make it easier to add it to the currency manager UI in the future if we wanted to.